File: src/main/java/de/dk/api/codeChallenge/notification/model/UserEntity.java
package de.dk.api.codeChallenge.notification.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import java.util.UUID;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "users")
@Getter
@Setter
@NoArgsConstructor
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "user_notifications", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "notification_type_id"))
    private Set<NotificationTypeEntity> notifications = new HashSet<>();

    public UserEntity(UUID id) {
        this.id = id;
    }

    public UserEntity(UUID id, Set<NotificationTypeEntity> notifications) {
        this.id = id;
        this.notifications = notifications != null ? notifications : new HashSet<>();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        UserEntity that = (UserEntity) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/kotlin/de/dkb/api/codeChallenge/notification/model/UserEntity.kt
package de.dkb.api.codeChallenge.notification.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "users")
class UserEntity(

    @Id
    @Column(columnDefinition = "uuid")
    val id: UUID = UUID.randomUUID(),

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "user_notifications",
        joinColumns = [JoinColumn(name = "user_id")],
        inverseJoinColumns = [JoinColumn(name = "notification_type_id")]
    )
    val notifications: MutableSet<NotificationTypeEntity> = mutableSetOf()
) {
    override fun equals(other: Any?): Boolean =
        this === other || (other is UserEntity && id == other.id)

    override fun hashCode(): Int = id.hashCode()
}